astropy/io/fits/file.py

```diff
@@ -400,6 +404,36 @@ def _overwrite_existing(self, overwrite, fileobj, closed):
             else:
                 raise IOError("File {!r} already exists.".format(self.name))
 
+    def _test_for_compression(self, obj_or_name, magic, mode, ext=None):
```

Comment:
Maybe `ext` should be `''` by default. On Python3 one can't compare strings to `None`.